ckan/cli/db.py

```diff
@@ -58,13 +58,34 @@ def clean():
 
 
 @db.command()
-@click.option(u'-v', u'--version', help=u'Migration version', default=u'head')
+@click.option('-v', '--version', help='Migration version', default='head')
+@click.option('--skip-plugins', is_flag=True, help='Skip plugin migrations')
+@click.option('--skip-core', is_flag=True, help='Skip core migrations')
+@click.pass_context
 @applies_to_plugin
-def upgrade(version: str, plugin: str):
+def upgrade(
+        ctx: click.Context, version: str, plugin: str,
+        skip_core: bool, skip_plugins: bool
+):
     """Upgrade the database.
     """
-    _run_migrations(plugin, version)
-    click.secho(u'Upgrading DB: SUCCESS', fg=u'green', bold=True)
+    if not skip_core:
+        _run_migrations(plugin, version)
+
+    if not skip_plugins and not plugin:
+        _migrate_plugins(apply=True)
```

Comment:
I'm sorry, but it's too opinionated suggestion. In the same way, someone can tell that it's hard to follow big number of private methods. You are suggesting introducing three private methods instead of one for calling two of them inside a third one. That's not ideal either. 

And I may be not the native speaker of English, but I'm sure that there are no double negations here. `not skip and not plugin` - that's two different negative assertions. `not skip` itself can hardly be callded double negation, because it's not the same as `not not_run`(which, of course, would be a double negation)

So your suggestion from my point of view doesn't make any good.